integration-tests/src/main/java/org/apache/polaris/service/it/ext/PolarisIntegrationTestExtension.java

```diff
@@ -0,0 +1,157 @@
+import org.apache.polaris.service.it.env.AuthToken;
+import org.apache.polaris.service.it.env.ClientCredentials;
+import org.apache.polaris.service.it.env.PolarisApiClient;
+import org.apache.polaris.service.it.env.PolarisFeatureConfig;
+import org.apache.polaris.service.it.env.Server;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.extension.ExtensionContext.Store.CloseableResource;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.platform.commons.util.AnnotationUtils;
+import org.junit.platform.engine.UniqueId;
+
+public class PolarisIntegrationTestExtension implements ParameterResolver {
+  private static final Namespace NAMESPACE =
+      Namespace.create(PolarisIntegrationTestExtension.class);
+
+  private static final PolarisServerManager manager =
+      ServiceLoader.load(PolarisServerManager.class)
+          .findFirst()
+          .orElseThrow(() -> new IllegalStateException("PolarisServerManager not found"));
+
+  @Override
+  public boolean supportsParameter(
+      ParameterContext parameterContext, ExtensionContext extensionContext)
+      throws ParameterResolutionException {
+    Class<?> type = parameterContext.getParameter().getType();
+    return type.isAssignableFrom(PolarisApiClient.class)
+        || type.isAssignableFrom(ClientCredentials.class)
+        || type.isAssignableFrom(AuthToken.class);
+  }
+
+  @Override
+  public Object resolveParameter(
+      ParameterContext parameterContext, ExtensionContext extensionContext)
+      throws ParameterResolutionException {
+    Env env = env(extensionContext);
+    Class<?> type = parameterContext.getParameter().getType();
+    if (type.isAssignableFrom(PolarisApiClient.class)) {
+      return env.client();
+    } else if (type.isAssignableFrom(ClientCredentials.class)) {
+      return env.server.adminCredentials();
+    } else if (type.isAssignableFrom(AuthToken.class)) {
+      return env.server.adminToken();
+    }
+    throw new IllegalStateException("Unable to resolve parameter: " + parameterContext);
+  }
+
+  private Env env(ExtensionContext context) {
+    ExtensionContext classCtx = classContext(context);
+    ExtensionContext.Store store = classCtx.getStore(NAMESPACE);
+    return store.getOrComputeIfAbsent(
+        Env.class,
+        (key) -> {
+          Class<?> testClass = classCtx.getRequiredTestClass();
+          Map<String, String> featureConfig =
+              AnnotationUtils.findRepeatableAnnotations(testClass, PolarisFeatureConfig.class)
+                  .stream()
+                  .collect(
+                      Collectors.toMap(PolarisFeatureConfig::name, PolarisFeatureConfig::value));
+          return new Env(manager.realm(), manager.startServer(featureConfig));
```

Comment:
I'm struggling to adapt this to Quarkus. In a Quarkus test (integration or not), the application will be running already when the extension is executed, and it's not possible to change the feature configs at that moment. The only way to change the application config is through a `QuarkusTestProfile` afaik. I don't think we can use a `QuarkusTestResourceLifecycleManager` to launch the application itself on demand; that's intended for dependent resources like databases etc. I think the `@PolarisFeatureConfig` annotation would eventually go away completely with Quarkus.

Reply:
Let me see if we this can be refactored to fit Quarkus better

Reply:
On a closer look at the test that use `@PolarisFeatureConfig` I tend to think that those tests are an overkill to have at the integration test level. The test variations merely check how configuration keys are resolved between catalog properties, server config and default values. I believe that part can be done by a unit test.

I'll try and refactor the blackbox tests to rely only on default config settings.